### DISCUSSION TITLE
Create an 'AFK' button for multiplayer rooms

### DISCUSSION THREAD

**Discussion by itsirfan05** [ppy/osu#27461] [Ideas]
This can be especially great for when you need to do something really quick that requires you to not be able to play the next beatmap, but not having to have to leave the multiplayer room. At times, I would have to use the 'spectate' button, to make myself disappear.

This feature request would allow players to know that the specific user is currently momentarily unavailable to play, rather than clicking the 'spectate' button, leaving the multiplayer, or being forced into the match.

**Comment by Rekunan** [comment]
Can I ask why using the spectate button is an issue for this?

**Reply by itsirfan05** [reply]
It's not an issue. It's much more convenient to have an 'AFK' button, rather than clicking the 'spectate' match button to not be forced going into the match without leaving the multiplayer room.

The 'AFK' button is self-explanatory. For examples; your parents call you and you need to do something, you got an important text message that you need to respond to, etc. The spectate button is for watching people play, rather than just being AFK. Maybe if you are AFK for more than 30 minutes, than multiplayer server/bancho can kick you out for being idle to long.

I guess the 'spectate' button is an AFK button, but imo it's better to have that feature rather than watching people's replays.

**Reply by Rekunan** [reply]
Hmm ok, what do u think should happen then if the user kept AFK on and then the host started the map.

**Reply by itsirfan05** [reply]
If the user presses the proposed 'AFK' button, then they will miss out the current round of the Multiplayer session. It will show (just like a current status) to everyone that that person is AFK. Meaning that they need to do something, wanna take a break, or don't wanna play the next map. This will keep happening until the person turns off 'AFK' mode.

If (as expressed) that a person is AFK for more than 30 minutes, then the multiplayer server will remove the person from the current room with a dialog stating "You've been removed from the multiplayer room for being AFK for more than 30 minutes!". This situation can be used to avoid hogging server resources.

**Reply by MeuhMeuh62** [reply]
i totally agree with this new feature, i hope that is implemented into the game
BUT, for next updates, i hope we implement tournament features and BANCHO pls!!
A referee like me won't this, but can be implemented with a mp command like `!mp afk off` for referees&hosts, and haunt it.

**Reply by Rekunan** [reply]
There are already plans to create tooling specifically for refs and other tourney staff, so that mp commands and other IRC interactions won't be necessary.
